tests/xdg_decoration.cpp

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright © 2020 Canonical Ltd.
+ *
+ * This program is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 3,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "in_process_server.h"
+#include "xdg_decoration_unstable_v1.h"
+#include "gmock/gmock.h"
+#include <gtest/gtest.h>
+
+using testing::_;
+using testing::NotNull;
+
+using namespace wlcs;
+
+namespace
+{
+auto const any_width = 300;
+auto const any_height = 300;
+
+struct ToplevelDecoration : StartedInProcessServer
+{
+    Client a_client{the_server()};
+    Surface a_surface{a_client.create_xdg_shell_stable_surface(any_width, any_height)};
```

Comment:
This incantation hides the `xdg_get_toplevel`. If you need to access that, there is `XdgSurfaceStable`. See, for example, `ForeignToplevelHandleTest`